### ISSUE TITLE
Add `torch.compile()` support to speed up inference

### ISSUE THREAD

**Issue by themattinthehatt** [danbider/lightning-pose#467]
## Background
 
Benchmarking showed `torch.compile()` gives solid inference speedups with a single-line change — up to ~6x on some model/GPU combinations (see `docs/source/increasing_inference_speed.rst`). We want to expose this through both the Python API and the CLI so users can opt into it without writing any PyTorch code themselves.
 
### 1. Add a `compile()` method to `Model` API
 
In `lightning_pose/api/model.py`, add a method that compiles the model's forward pass in place:
 
```python
def compile(self) -> None:
    """Compile the model's forward pass with torch.compile() for faster inference.
 
    Must be called before running prediction. Compilation is triggered lazily on
    the first inference call (can take tens of seconds); subsequent calls with the
    same input shape reuse the compiled graph. Changing batch size or input
    resolution triggers a new compilation automatically.
    """
    self.model.forward = torch.compile(self.model.forward)
```
 
**Important implementation detail:** compile `self.model.forward` directly and reassign it, rather than wrapping the whole module (`self.model = torch.compile(self.model)`). Some of our prediction code paths call `get_loss_inputs_labeled` rather than going through `__call__`/`forward`, and compiling the whole module silently falls back to the *original* uncompiled submodule for any method other than `forward`/`__call__`. Compiling `forward` directly avoids this trap. (See the reference doc linked above for more detail on why this matters.)
 
### 2. Usage pattern — Python API
 
```python
from lightning_pose.api import Model
 
model = Model.from_dir("path/to/model_dir")
model.compile()
result = model.predict_on_video_file("path/to/video.mp4")
```
 
### 3. Add a `--compile` flag to the CLI's `predict` command
 
```bash
litpose predict /path/to/model_dir /path/to/video.mp4 --compile
```
 
Wire this up where `litpose predict` constructs its `Model` and calls `predict_on_video_file`/equivalent — after loading the model, call `.compile()` if the flag is set:
 
```python
parser.add_argument(
    "--compile",
    action="store_true",
    default=False,
    help="Compile the model with torch.compile() for faster inference.",
)
 
# ...
model = Model.from_dir(args.model_dir)
if args.compile:
    model.compile()
result = model.predict_on_video_file(args.video_path)
```
 
## Testing
 
- Add a unit test that calls `model.compile()` and runs a prediction on a labeled csv, asserting it completes without error and returns predictions of the expected shape. Add another unit tests that runs a prediction on the short test video fixture. Both unit tests can just be for a singleview model.
- Add an accuracy test: compare predictions from a compiled model against the same eager model on a few real frames (label csv file), and assert max pixel deviation is small (e.g. under 0.1px) — consistent with the numbers already in the benchmarking doc. This guards against a future PyTorch version changing compile behavior in a way that actually affects outputs. This test only needs to run on a singleview model.
- Test that calling `.compile()` twice, or compiling then running on a different batch size, doesn't error (should just trigger recompilation).
- Add a CLI test that `--compile` runs end-to-end on a small test video.

## Documentation
 
- Update the existing inference speed doc, showing the `model.compile()` and `litpose predict --compile` usage patterns above.
- Note that the first call after compiling will be slower (compilation overhead) and this is expected.

## Out of scope for this issue
 
This issue is scoped to `torch.compile()` only. Please don't add ONNX/TensorRT export or loading — that will be handled as a separate, follow-up piece of work.
